src/solvers/td3.ml

```diff
@@ -508,16 +508,17 @@ module WP =
               (* collect function return for reluctant analysis *)
               mark_node obsolete_ret f (Function f)
           ) changed_funs;
-        (* Unknowns from partially changed functions need only to be collected for eager destabilization when reluctant is off *)
+        (* Primary changed unknowns from partially changed functions need only to be collected for eager destabilization when reluctant is off *)
+        (* The return nodes of partially changed functions are collected in obsolete_ret for reluctant analysis *)
         (* We utilize that force-reanalyzed functions are always considered as completely changed (and not partially changed) *)
-        if not reluctant then (
-          List.iter (fun (f, pn, _) ->
+        List.iter (fun (f, pn, _) ->
+            if not reluctant then (
               List.iter (fun n ->
                   mark_node obsolete_prim f n
                 ) pn;
-              mark_node obsolete_ret f (Function f);
-            ) part_changed_funs;
-        );
+            );
+            mark_node obsolete_ret f (Function f);
```

Comment:
Could this line be in the `else`-clause? `obsolete_ret` and `old_ret` are only accessed when `reluctant` is on.

Reply:
I think that should be possible.